includes/class-wc-stripe-blocks-support.php

```diff
@@ -308,15 +308,24 @@ public function add_stripe_intents( PaymentContext $context, PaymentResult &$res
 				|| ! empty( $result->payment_details['setup_intent_secret'] )
 			)
 		) {
-			$payment_details                          = $result->payment_details;
-			$payment_details['verification_endpoint'] = add_query_arg(
+			$payment_details       = $result->payment_details;
+			$verification_endpoint = add_query_arg(
 				[
 					'order'       => $context->order->get_id(),
 					'nonce'       => wp_create_nonce( 'wc_stripe_confirm_pi' ),
 					'redirect_to' => rawurlencode( $result->redirect_url ),
 				],
 				home_url() . \WC_Ajax::get_endpoint( 'wc_stripe_verify_intent' )
 			);
+
+			if ( isset( $payment_details['save_payment_method'] ) && ! empty( $payment_details['save_payment_method'] ) ) {
```

Comment:
I think it'd be enough to just check

```suggestion
			if ( ! empty( $payment_details['save_payment_method'] ) ) {
```

According to [the PHP documentation](https://www.php.net/manual/en/function.empty.php):

> That means empty() is essentially the concise equivalent to `! isset( $var ) || $var == false`